addons/udes_stock/models/stock_picking.py

```diff
@@ -627,25 +635,37 @@ def _create_batch(self, pickings):
         PickingBatch = self.env["stock.picking.batch"]
         PickingBatch.create({"picking_ids": [(6, 0, pickings.ids)]})
 
-    def _prepare_move(self, pickings, products_info, **kwargs):
+    def _prepare_move(self, pickings, products_info, extra_product_info_keys=None, **kwargs):
```

Comment:
Can we handle it with super and set a context in case we want to set additional fields on the move when calling create_picking?

The change do not seem very generic to add in udes open, and increases complexity to my opinion

Reply:
the change would be mostly the same with context as the change is in the inner loop.

We lose information by calling super and modifying the result externally (i tried this initially, but the relation of products_info and pickings and how they end up in the result of this function is confusing)

Reply:
I don't understand what is not generic about this change

Reply:
Suggested below a solution i think

Reply:
It becomes very complex to remember to follow the code 
product_info, extra_product_info_keys , and changing kwargs with product_info[key] if key in extra_product_info_keys. I believe the suggested way to get the move vals into a method would be easier to override with super and of course less code change

Reply:
c8075a7402d85b91ecdb40b94a78459c3fb04a9e